### ISSUE TITLE
Bring back PWA manifest + service worker

### ISSUE THREAD

**Issue by romannurik** [romannurik/SlidesCodeHighlighter#88]

**Comment by srivats22** [comment]
I would be interested in helping out with this...

I made the changes locally attaching the screenshots if this is what you are expecting then I will go ahead and open a PR for it

# Install Popup
<img width="1362" alt="Screenshot 2024-11-01 at 7 59 37 AM" src="https://github.com/user-attachments/assets/cc205f2b-aa18-4f53-8463-9da7b0697b29">

# Running as a PWA
<img width="1189" alt="Screenshot 2024-11-01 at 7 59 51 AM" src="https://github.com/user-attachments/assets/ee6386ae-ebe5-442b-822c-e2913d2c01f6">

# Code also works
<img width="1425" alt="Screenshot 2024-11-01 at 8 00 45 AM" src="https://github.com/user-attachments/assets/4272905c-4368-43e7-8fa9-7f1725faf5a8">

# Manifest being read
<img width="579" alt="Screenshot 2024-11-01 at 8 06 31 AM" src="https://github.com/user-attachments/assets/1b4dcd94-b12a-46a1-9fd5-b34b2418781e">

Thanks

**Comment by romannurik** [comment]
Hey there, that's awesome, thank you! I'm curious how large the changes were .. im hoping most of the heavy lifting can be done by a vite plugin? And yes feel free to open a PR :-)

**Comment by srivats22** [comment]
Yes most of it was from the vite plugin I will go ahead and open one right now and you can review the changes

**Comment by romannurik** [comment]
Fixed in 5871dff66d46888d53eaaed29fafcaf67501ff60
